lib/core/security/roleRepository.js

```diff
@@ -446,25 +441,12 @@ class RoleRepository extends Repository {
               didYouMean(action, plugins.getActions(roleController)));
           }
 
-          this.kuzzle.log.warn(`The role "${role._id}" gives access to the non-existing action "${action}" for the controller "${roleController}".`);
+          this.kuzzle.log.warn(`[SECURITY] The role "${role._id}" gives access to the non-existing action "${action}" for the controller "${roleController}".`);
         }
       });
     });
   }
 
-  /**
-   * Fetching roles and check for each of them for invalid plugin rights.
-   * If there are some, Kuzzle will log a warning.
-   */
-
-  async sanityCheck () {
```

Comment:
Why do you remove the sanity check? It prints only warnings about potentially unused plugin rights, to allow maintainers to clean up their projects?

Reply:
Because when we register the rights we already print warnings about rights that uses non-existent controllers.

But you're right this is useful so I will modify this PR to not print warning when registering rights and I will run the sanity check after plugins loading :+1: